pkg/framework/aries/api/factory.go

```diff
@@ -0,0 +1,16 @@
+/*
+Copyright SecureKey Technologies Inc. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package api
+
+import (
+	"github.com/hyperledger/aries-framework-go/pkg/didcomm/protocol/exchange"
+)
+
+// ProtocolProviderFactory allows overriding of aries protocol providers
+type ProtocolProviderFactory interface {
+	CreateExchangeProvider() exchange.Provider
```

Comment:
This is not the right abstraction. There cannot be a dependency back to "github.com/hyperledger/aries-framework-go/pkg/didcomm/protocol/..."

Reply:
You should be creating generic factories/providers here - like Transports.

Reply:
[Updated](https://github.com/hyperledger/aries-framework-go/compare/f4a9ffc938e46fb640514954eef882c1518e1871..463122c87b11e27a303f59b42906628162e71b73#diff-9988f5820bf05228e5657f2842fa071eR16) the the abstraction.